platform/darwin/src/MGLShapeSource.h

```diff
@@ -321,6 +323,49 @@ MGL_EXPORT
  */
 - (NSArray<id <MGLFeature>> *)featuresMatchingPredicate:(nullable NSPredicate *)predicate;
 
+/**
+ Returns an array of map features that are the leaves of the specified cluster.
+ ("Leaves" are the original points that belong to the cluster.)
+ 
+ This method supports pagination; you supply an offset (number of features to skip)
+ and a maximum number of features to return.
+ 
+ @param cluster An object that conforms to the `MGLCluster` protocol. Currently
+    the only types that can conform are private subclasses of `MGLPointFeature`.
+ @param offset Number of features to skip.
+ @param limit Maximum number of features to return
+ 
+ @return An array of objects that conform to the `MGLFeature` protocol.
+ */
+- (NSArray<id <MGLFeature>> *)leavesOfCluster:(id<MGLCluster>)cluster offset:(NSUInteger)offset limit:(NSUInteger)limit;
+
+/**
+ Returns an array of map features that are the immediate children of the specified
+ cluster *on the next zoom level*. The may include features that also conform to
+ the `MGLCluster` protocol.
+ 
+ @param cluster An object that conforms to the `MGLCluster` protocol. Currently
+    the only types that can conform are private subclasses of `MGLPointFeature`.
```

Comment:
I’m not sure developers intuitively understand the implications of a private subclass. How about:

> This object should be an object returned from a feature querying method, such as `-[MGLMapView visibleFeaturesInRect:]` or `-[MGLShapeSource featuresMatchingPredicate:]`. Do not pass in a custom class that conforms to this protocol.